qiita_pet/templates/search_sample_list.html

```diff
@@ -0,0 +1,65 @@
+<!--Study datatypes tabs-->
+{% from future.utils import viewitems %}
+{% from qiita_pet.util import clean_str %}
+{% set sel = "" %}
+{% if selected is True %}
+    {% set sel = "-sel" %}
+  {% end %}
+
+<ul class="nav nav-tabs" role="tablist" id="dt-tabs-{{sid}}{{sel}}">
+{% for data_type in proc_data_dict %}
+  <li class="active"><a href="#{{sid}}-{{data_type}}{{sel}}" role="tab" data-toggle="tab">{{data_type}}</a></li>
+{% end %}
+</ul>
+
+{% for data_type, proc_data in viewitems(proc_data_dict) %}
+<div class="tab-content" id="pane-{{sid}}-{{data_type}}{{sel}}">
+  <!--Study datatype processed data tabs-->
+  <ul class="nav nav-tabs" role="tablist" id="dt-tabs-{{sid}}-proc-{{sel}}">
+  {% for pid in proc_data %}
+    <li class="active"><a href="#pane-{{sid}}-{{data_type}}-{{pid}}{{sel}}" role="tab" data-toggle="tab">Proc Data {{pid}}</a></li>
+  {% end %}
+  </ul>
+{% end %}
+
+<!--Sample divs per datatype per processed data id-->
+{% for data_type, proc_data in viewitems(proc_data_dict) %}
+  {% for pid in proc_data %}
+  <div class="tab-content" id="pane-{{sid}}-{{data_type}}-{{pid}}{{sel}}">
+    <!--samples list-->
+    <h3>Samples</h3>
+    <a href="#" onclick="select_deselect({{sid}}, {{pid}}, '{{sel}}', true); return false;">Select all</a> | <a href="#" onclick="select_deselect({{sid}}, {{pid}}, '{{sel}}', false); return false;">Select none</a> | <a href="#" onclick="select_inverse({{sid}}, {{pid}}, '{{sel}}'); return false;">Select inverse</a>
+  <table class='table table-hover'>
+    <tr>
+      <th></th>
+      <th></th>
+      <th>Sample ID</th>  
+    {% for meta in meta_headers %}
+      <th>{{meta}}</th>
+    {% end %}
+    </tr>
+    {% if meta_headers %}
+      {% for sample in proc_data_samples[pid]%}
+    <tr><td>
+      <input type='checkbox' name='samples{{sel}}'
+      value='{{pid}}#{{sample[0]}}' class='study{{sid}} proc{{pid}} {{ " ".join(clean_str(s) for s in sample) }} {{sel}}' onchange='count_update({{sid}})'></td>
+      <td><a href="#" onclick="select_category('{{clean_str(sample[0])}}', '')">All Datatypes</a></td>
+        {% for val in sample %}
```

Comment:
Let's run through with a small example, say, 10 studies, each with 2 processed IDs, each with 100 samples, and each with 20 metadata values. This would result in 40,000 executions of the inner loop, which is building a large HTML table (without knowing the internals of the templating engine, this may be very expensive to do). In addition to the HTML table, the 2nd nesting will end up doing 44,000 regex calls (lines 45, 57 and 58). In other words, this quickly gets very expensive especially if we have concurrent users. 

Let's further imagine that the compilation of this page takes 100ms. This is a reasonable estimate given that doing a (naive) regex 44,000 times takes ~50ms:

```python
> %%timeit
for i in range(44000):
  re.search('foo', 'asdasdasdasdasdasda')
   ....:
10 loops, best of 3: 53.5 ms per loop
```

What that means is that, with 10 concurrent users doing similar small searches, the server is blocked for a full second. This will greatly impact the user experience. There is also the potential for a single user to form a search that grinds the site to a full halt for an extended period. An expense here limits our ability to scale to concurrent users among other variables. 

For something like search, we very likely want to cache and index (external to postgres) aggressively. Given that we're approaching alpha, I think we probably want to hold off on a caching exploration, but it should be something we discuss soon.